client_connection/client_connection.go

```diff
@@ -0,0 +1,55 @@
+package client_connection
+
+import (
+	"errors"
+	"github.com/mysterium/node/service_discovery/dto"
+)
+
+type State string
+
+const (
+	NOT_CONNECTED = State("NOT_CONNECTED")
+	NEGOTIATING   = State("NEGOTIATING")
+	CONNECTED     = State("CONNECTED")
+)
+
+var (
+	ALREADY_CONNECTED = errors.New("already connected")
+)
+
+type Status struct {
```

Comment:
`Status` and `State` looks confusing. What about ranaming `Status` -> `Connection`?

Reply:
Makes sense. But Status is object which represents connection state and possible other information like bytes sent/received last error etc. My suggestion Status -> ConnectionStatus